diagnostics.cpp

```diff
@@ -7,11 +7,11 @@
  */
 void DeRestPluginPrivate::handleDiagnosticsClusterIndication(const deCONZ::ApsDataIndication &ind, deCONZ::ZclFrame &zclFrame)
 {
-    Sensor *sensor = getSensorNodeForAddressAndEndpoint(ind.srcAddress(), ind.srcEndpoint());
+    Sensor *sensor = getSensorNodeForAddressAndEndpoint(ind.srcAddress(), ind.srcEndpoint(), QLatin1String("ZHAThermostat"));
 
     if (!sensor)
     {
-        DBG_Printf(DBG_INFO, "No sensor found for 0x%016llX, endpoint: 0x%08X\n", ind.srcAddress().ext(), ind.srcEndpoint());
+        DBG_Printf(DBG_INFO, "No thermostat sensor found for 0x%016llX, endpoint: 0x%08X\n", ind.srcAddress().ext(), ind.srcEndpoint());
```

Comment:
`handleDiagnosticsClusterIndication()` isn't this to be supposed to be general, why thermostat?

`endpoint: 0x%08X` shouldn't this be `endpoint: 0x%02X`?

Reply:
For the endpoint: Right, corrected that in other files as well

For the generic handling: Happy to remove the "thermostat" from the message. Regarding the generic usage, currently only 2 thermostats require this cluster. However, we allow further processing only if a sensor is available. And to pick the right one, I currently have to pin the sensor type some lines above.